google-cloud-trace/lib/google/cloud/trace/faraday_middleware.rb

```diff
@@ -43,17 +43,34 @@ def add_request_labels span, env
           label_keys = Google::Cloud::Trace::LabelKey
 
           set_label labels, label_keys::HTTP_METHOD, env.method
-          set_label labels, label_keys::HTTP_USER_AGENT,
-                    env.request_headers[:user_agent]
           set_label labels, label_keys::HTTP_URL, env.url.to_s
+
+          # Only sets request size if request is not sent yet.
+          unless env.status
+            request_body = env.body || ""
+            set_label labels, label_keys::RPC_REQUEST_SIZE,
+                      request_body.bytesize.to_s
+          end
         end
 
         ##
         # @private Set Trace span labels from response
-        def add_response_labels span, response
-          set_label span.labels,
-                    Google::Cloud::Trace::LabelKey::HTTP_STATUS_CODE,
-                    response.status.to_s
+        def add_response_labels span, env
+          labels = span.labels
+          label_keys = Google::Cloud::Trace::LabelKey
+
+          response = env.response
+          response_body = env.body || ""
```

Comment:
I don't think `env.request.body` is a thing. Request body is submitted as `env.body`. Yeah, let me change it to `env.response.body` here for the peace of mind.